lib.py

```diff
@@ -120,8 +125,13 @@ async def build_code(author_name: str, author_id: int, tmp_dir: str) -> bool:
             remove=True,
             stdout=True,
             mem_limit="8g",
-            # network_mode="none", # Want no-network, but it's downloading crates. :(
-            volumes={tmp_dir: {'bind': '/app', 'mode': 'rw'}}
+            network_mode="none",
+            volumes={
+                os.path.join(tmp_dir, "src"): {'bind': '/app/src', 'mode': 'rw'},
+                os.path.join(tmp_dir, "benches"): {'bind': '/app/benches', 'mode': 'rw'},
+                os.path.join(tmp_dir, "inputs"): {'bind': '/app/inputs', 'mode': 'rw'},
+                os.path.join(tmp_dir, "target"): {'bind': '/app/target', 'mode': 'rw'},
+                }
```

Comment:
is /app/inputs required for building?

Reply:
Seemingly not. There's no harm in having the extra volume; the temp-dir source for the volume won't have any input files in it until later. That said, there's also no reason to mount the volume now, so that line is deleted now.